### PR TITLE
Add option to only show searchfield during search

### PR DESCRIPTION
**Pull request by weisJ**
A possible implementation of #831 

Settings:
image

### CODE REVIEW

File: Maccy/Observables/AppState.swift [hunk 1]
```diff
@@ -39,6 +40,17 @@ class AppState: Sendable {
     }
   }
 
+  var searchVisible: Bool {
```
[the remark this answers is not in the document]
Comment (weisJ) [reply Maccy/Observables/AppState.swift:43]: It‘s a remnant of another experiment I did. Will change later.

[the remark this answers is not in the document]
Comment (weisJ) [reply Maccy/Observables/AppState.swift:40]: Done

[the remark this answers is not in the document]
Comment (weisJ) [reply Maccy/Observables/AppState.swift:40]: Now really fixed.

**Comment by p0deje** [comment]
@weisJ Any chance you could re-open this to `master` branch?

**Comment by weisJ** [comment]
Sure thing. Opened #870 targeting `master`

**Pull request closed without merging**